File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] result = {0, 0};
        
        int area = brown + yellow;
        for (int w = 3; w <= area / 3; w++) {
            int h = area / w;
            if (area % w != 0 || w < h) continue;
            
            int y = (w - 2) * (h - 2);
            int b = area - y;
            
            if (b == brown && y == yellow) {
                result[0] = w;
                result[1] = h;
                break;
            }
        }
        
        return result;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    area = brown + yellow
    
    for w in range(3, area // 3 + 1):
        h = area // w
        
        if area % w != 0 or w < h:
            continue
                
        brown_cnt = (w + h) * 2 - 4
        yellow_cnt = area - brown_cnt
        
        if brown_cnt == brown and yellow_cnt == yellow:
            return [w, h]